akka-actor/src/main/scala/akka/actor/dungeon/DeathWatch.scala

```diff
@@ -56,14 +56,14 @@ private[akka] trait DeathWatch { this: ActorCell =>
           watching = removeFromMap(a, watching)
         }
       }
-      terminatedQueued = removeFromSet(a, terminatedQueued)
+      terminatedQueued = removeFromMap(a, terminatedQueued)
       a
   }
 
   protected def receivedTerminated(t: Terminated): Unit =
-    if (terminatedQueued(t.actor)) {
+    terminatedQueued.get(t.actor).foreach { optionalMessage ⇒
       terminatedQueued -= t.actor // here we know that it is the SAME ref which was put in
```

Comment:
was this the leak? when it was a custom message the `receivedTerminated` was never invoked and therefore the `terminatedQueued` wasn't cleared

Reply:
Yep! Thanks for the quick review!